### ISSUE TITLE
[Feature Request] ScnRev: Carrier + Timeout

### ISSUE THREAD

**Issue by VA7FGT** [AubsUK/uv-k5-firmware-custom#25]
When scanning through a list of channels it's fairly normal to come across a back-and-forth that will break carrier and continue scanning. 

A configurable delay to linger on a channel after carrier breaks before continuing scanning would be handy, even moreso if the timer reset once carrier was seen again. 

Extra extra handy would be a max-total-linger parameter as well. So you don't get stuck on the same channel for a few minutes without exiting scan mode :P

**Comment by AubsUK** [comment]
I believe that's what Menu 21 `ScnRev` is for (same as Egzumer's original Menu 22 [egzumer/wiki/Menu](https://github.com/egzumer/uv-k5-firmware-custom/wiki/Menu))

CARRIER - resume scan after signal disappears
TIMEOUT - resume scan after 5 seconds pause
STOP - after receiving a signal, stop the scan

**Comment by VA7FGT** [comment]
Yeah, the issue is that in ```Carrier``` it will move off as soon as the signal is gone. ```Timeouts``` timer is set from the moment it's encountered and never refreshed. What I'm proposing here is a hybrid of the two :P

**Comment by AubsUK** [comment]
I think it should wait 5 seconds from the end of the last transmission, so if a new transmission is picked up on the same memory channel, the counter should restart when that carrier drops.  I think that would be simpler and consume less code than having multiple options.

**Comment by VA7FGT** [comment]
That seems like a simple and sensible course of action!

**Comment by OneBangDash** [comment]
I have ScnRev set to CARRIER but it continues scanning even during an active transmission. This is causing the audio to "skip" as it jumps between scanning and playing the transmission. I just updated to v0.07 from v0.02 and didn't have this issue until now.

**Comment by AubsUK** [comment]
> I just updated to v0.07 from v0.02 and didn't have this issue until now.

That's really interesting, I'll look at what the differences are between the two versions and what I can do about it.  Thank you for identify this.

**Comment by AubsUK** [comment]
@OneBangDash I've tried with v00.02 and v00.07, I get exactly the same performance.  The underlying Egzumer firmware also has the same issue. This has been discussed in the original repo for Egzumer firmware Stuttering audio on Narrow FM https://github.com/egzumer/uv-k5-firmware-custom/issues/467

I think what's happening is when the signal drops briefly, it sees the break and starts the countdown timer to continue scanning.  If the transmission returns, it should reset the countdown, but I don't think it is.  I'll look into how I can get it to do that, and also if I can identify why some signals drop while others don't (based on the above link, it might be to do with the Narrow FM setting).

**Comment by mlapalme99** [comment]
This would be amazing if the carrier timeout could be configured
